README.md

```diff
@@ -1,125 +1,148 @@
 
-Create a pod in the cluster 'test-cluster', an `IngressListener` for the app 'test-app' and an `EgressListener` that allows
-'test-app' to query `test-cluster:8080` by making requests to envoy at `127.0.0.1:9090`.
-```
-$ cd manifests/examples
-$ kubectl --context=exp-1-aws -n labs apply -f .
 ```
+    [
+      {
+        "name": "aws",
+        "kubeconfig": "/etc/aws/aws.kubeconfig",
+        "listenersource": true
+      },
+      {
+        "name": "gcp",
+        "kubeconfig": "/etc/gcp/gcp.kubeconfig",
+        "listenersource": false
+      }
+    ]
 
-You should see events registered in the logs of the control plane
-```
-2019/06/27 09:16:45 [DEBUG] received ADDED event for cluster test-cluster ip: 10.2.7.139
-```
-```
-2019/06/27 09:20:45 [DEBUG] received ADDED event for ingress listener example-listener: 0.0.0.0:8080 -> 127.0.0.1:8081
-2019/06/27 09:20:45 [DEBUG] Updating snapshot for node test-app
-2019/06/27 09:20:45 [DEBUG] Response: responding (2) with type: type.googleapis.com/envoy.api.v2.Cluster, version: 2019-06-27 01:20:45.705178 +0100 BST m=+240.565251561, resources: 1
-2019/06/27 09:20:45 [DEBUG] Response: responding (1) with type: type.googleapis.com/envoy.api.v2.Listener, version: 2019-06-27 01:20:45.705185 +0100 BST m=+240.565257900, resources: 1
 ```
 
-Check `http://localhost:9901/config_dump` to see the listeners and clusters that have been added to Envoy by the control plane.
+You need to supply a list of source and the location of the respective kubeconfig files.
+`listenersource` defines the primary kubernetes cluster, meaning the cluster which the control plane will watch for listeners. It should be set to true only for 1 cluster, the same one that the control plane is deployed. That way envoy sidecars in each kubernetes cluster will talk to the local control plane to fetch config, listeners will only be defined in the same cluster where the envoy sidecar is running and clusters will be generated from all the available sources.
+Losing networking between clusters means that remote cluster members will disappear but everything will keep working for the local cluster.
 
-Add another pod, to gcp this time, to see the cluster update:
-```
-$ kubectl --context=exp-1-gcp -n labs apply -f clusterpod.yaml
-```
+## Dynamic Configuration - envoy
+
+Configuration for listeners is generated via 2 kubernetes custom resources.
 
-## How it works
 ### IngressListener
-```yaml
+
+```
 apiVersion: ingresslistener.envoy.uw.systems/v1alpha1
 kind: IngressListener
 metadata:
-  name: example-listener
+  name: <name>
 spec:
-  nodename: test-app
-  listenport: 8080
-  targetport: 8081
-  rbacallowcluster: test-cluster
+  nodename: <envoy-node-name>
+  listenport: 8080 # Port to listen for ingress traffic
+  targetport: 80 # Port to forward on localhost
+  rbac:
+    cluster: <cluster> # Only accept incoming traffic from cluster
```

Comment:
Not really a review of this PR but `cluster` probably needs to be a list, to support multiple upstream services.